rust/kcl-lib/src/execution/types.rs

```diff
@@ -2113,4 +2113,46 @@ b = 180 / PI * a + 360
         // TODO type is not ideal
         assert_value_and_type("b", &result, 417.0, NumericType::radians());
     }
+
+    #[tokio::test(flavor = "multi_thread")]
+    async fn coerce_nested_array() {
+        let mut exec_state = ExecState::new(&crate::ExecutorContext::new_mock().await);
+
+        let mixed1 = KclValue::MixedArray {
+            value: vec![
+                KclValue::Number {
+                    value: 0.0,
+                    ty: NumericType::count(),
+                    meta: Vec::new(),
+                },
+                KclValue::Number {
+                    value: 1.0,
+                    ty: NumericType::count(),
+                    meta: Vec::new(),
+                },
+                KclValue::HomArray {
+                    value: vec![
+                        KclValue::Number {
+                            value: 2.0,
+                            ty: NumericType::count(),
+                            meta: Vec::new(),
+                        },
+                        KclValue::Number {
+                            value: 3.0,
+                            ty: NumericType::count(),
+                            meta: Vec::new(),
+                        },
+                    ],
+                    ty: RuntimeType::Primitive(PrimitiveType::Number(NumericType::count())),
+                },
+            ],
+            meta: Vec::new(),
+        };
+
+        // Principal types
+        let tym1 = RuntimeType::Union(vec![RuntimeType::Primitive(
```

Comment:
I think this should be an array type rather than a union type?

Reply:
sorry fixed i did this in a sleep deprived supor